codeBlock-1984/bankr
# #165412291 Setup and configure database (db)

### What does this PR do?
Adds a Postgres database to the project for persisting data

### Description of Task to be completed
- Create database instance on ElephantSQL and copy url
- Install pg as dependency
- Set database url to an env variable in .env
- Create dbConfig file and set up env configuration inside it
- Create database folder for database files
- Create db file and configure database instance using url
- Create Migration, models, and seeders files inside the database folder
- Write table creation queries in model files
- Create migration, migration:undo and seed scripts in package.json file
- Run migration script to create database tables
- Run seed script to seed the database

### How should this be manually tested?
1. Set up and configure the project locally, following the instructions in the README file
2. Create a Postgres database on ElephantSQL and get the connection url
3. Add the connection url as value to keys *DB_ENV* and *DB_DEV* in the local .env file created in step 1 above
4. Run ```npm run migrate && npm run seed```
5. Open your database in your terminal or any other application
6. You should see the tables created in your database along with seed data

### What are the relevant pivotal tracker stories?
#165412291
